src/axom/primal/operators/detail/winding_number_3d_memoization.hpp

```diff
@@ -0,0 +1,296 @@
+    //  since the expanded portions are never visibile
+    m_oBox = m_alteredPatch.orientedBoundingBox();
+    m_bBox.clear();
+    for(int i = 0; i < num_knot_span_u; ++i)
+    {
+      for(int j = 0; j < num_knot_span_v; ++j)
+      {
+        if(split_patches(i, j).getNumTrimmingCurves() == 0)
+        {
+          continue;  // Skip patches with no trimming curves
+        }
+
+        BezierPatch<T, 3> the_patch;
+        if(m_alteredPatch.isRational())
+        {
+          the_patch = BezierPatch<T, 3>(split_patches(i, j).getControlPoints(),
+                                        split_patches(i, j).getWeights(),
+                                        split_patches(i, j).getDegree_u(),
+                                        split_patches(i, j).getDegree_v());
+        }
+        else
+        {
+          the_patch = BezierPatch<T, 3>(split_patches(i, j).getControlPoints(),
+                                        split_patches(i, j).getDegree_u(),
+                                        split_patches(i, j).getDegree_v());
+        }
+
+        BezierPatch<T, 3> p1, p2, p3, p4;
+        the_patch.split(0.5, 0.5, p1, p2, p3, p4);
+
+        m_bBox.addBox(p1.boundingBox());
+        m_bBox.addBox(p2.boundingBox());
+        m_bBox.addBox(p3.boundingBox());
+        m_bBox.addBox(p4.boundingBox());
+      }
+    }
+
+    m_alteredPatch.expandParameterSpace(0.05, 0.05);
+
+    m_curveQuadratureMaps.resize(m_alteredPatch.getNumTrimmingCurves());
+  }
+
+  NURBSPatchGWNCache(const BezierPatch<T, 3> a_patch)
+    : NURBSPatchGWNCache(NURBSPatch<T, 3>(a_patch))
+  { }
+
+  // Mirror the functionality of NURBSPatch so signatures match in GWN evaluation.
+  // Allowing only access ensures the memoized information is always accurate
+  auto getControlPoints() const { return m_alteredPatch.getControlPoints(); }
+  auto getNumControlPoints_u() const { return m_alteredPatch.getNumControlPoints_u(); }
+  auto getNumControlPoints_v() const { return m_alteredPatch.getNumControlPoints_v(); }
+  auto getWeights() const { return m_alteredPatch.getWeights(); }
+  auto getKnots_u() const { return m_alteredPatch.getKnots_u(); }
+  auto getKnots_v() const { return m_alteredPatch.getKnots_v(); }
+  auto getMinKnot_u() const { return m_alteredPatch.getMinKnot_u(); }
+  auto getMaxKnot_u() const { return m_alteredPatch.getMaxKnot_u(); }
+  auto getMinKnot_v() const { return m_alteredPatch.getMinKnot_v(); }
+  auto getMaxKnot_v() const { return m_alteredPatch.getMaxKnot_v(); }
+  auto getTrimmingCurves() const { return m_alteredPatch.getTrimmingCurves(); };
+  auto getNumTrimmingCurves() const { return m_alteredPatch.getNumTrimmingCurves(); }
+  auto getParameterSpaceDiagonal() const { return m_pboxDiag; }
+
+  // Access precomputed data
+  auto getAverageNormal() const { return m_averageNormal; }
+  auto boundingBox() const { return m_bBox; }
+  auto orientedBoundingBox() const { return m_oBox; }
```

Comment:
Do we want to return copies? or const references?

Reply:
I suppose this should change if/when the corresponding functions in `NURBSPatch` change...

Reply:
Sounds like something to look into in the future